lib/stat_tracker.rb

```diff
@@ -108,4 +108,26 @@ def lowest_total_score
     end
     scores.min
   end
+  def coach_win_percentages
+    coach_games = Hash.new { |hash, key| hash[key] = { wins: 0, games: 0}}
+    @all_game_teams.each do |game_team|
+      coach = game_team.head_coach
+      coach_games[coach][:games] += 1
+      coach_games[coach][:wins] += 1 if game_team.result == "WIN"
+    end
+      coach_games.transform_values do |stats| 
+        games = stats[:games]
+        games > 0 ? ((stats[:wins].to_f / games) * 100).round : 0
+      end
+  end
+
+  def winningest_coach
+    coach_win_percentages = coach_win_percentages
+    coach_win_percentages.max_by { |coach, win_percentage| win_percentage}.first
+  end
+
+  def worst_coach
+    coach_win_percentages = coach_win_percentages()
+    coach_win_percentages.min_by { |coach, win_percentage| win_percentage}.first
```

Comment:
in the event of a tie, you're going to return the first coach. Is this intended?

Reply:
yes because it will show the min coach first